bin/utils.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @typedef { import("../types").SanityBuildPluginEntity } SanityBuildPluginEntity
+ * @typedef { import("../types/plugins").SanityPluginLookup } SanityPluginLookup
+ * @typedef { import("../types/plugins").BuildPluginEntity } BuildPluginEntity
+ */
+
+// TODO: remove this an use assert.deepEqual once we support only node engines > 18
+import deepEqual from 'deep-equal'
+
+const sanityFieldNameToPluginKeyLookup = {
+  _id: '_id',
+  title: 'name',
+  description: 'description',
+  // In sanity, the field is an array of authors, in plugins.json. it's one author
+  // authors: 'author',
+  packageName: 'package',
+  repoUrl: 'repo',
+  version: 'version',
+  // an object that can be null
+  compatibility: 'compatibility',
+}
+
+const pluginKeyToSanityFieldNameLookup = Object.entries(sanityFieldNameToPluginKeyLookup).reduce(
+  (lookup, [key, value]) => {
+    // eslint-disable-next-line no-param-reassign
+    lookup[value] = key
+
+    return lookup
+  },
+  {},
+)
+
+/**
+ * Retrieves a list of all the plugins stored in Sanity
+ *
+ * @param plugins {SanityBuildPluginEntity[]} Query to execute
+ * @returns A list of all the plugins stored in Sanity
+ */
+export const getSanityPluginLookup = (plugins) => {
+  /**
+   * @type {SanityPluginLookup}
+   */
+  const pluginLookup = plugins.reduce((sanitytPluginLookup, plugin) => {
+    // eslint-disable-next-line no-param-reassign
+    sanitytPluginLookup[plugin.packageName] = plugin
+
+    return sanitytPluginLookup
+  }, {})
+
+  return pluginLookup
+}
+
+/**
+ *
+ * @param {BuildPluginEntity} plugin
+ *
+ * @returns {SanityBuildPluginEntity}
+ */
+const convertToSanityPlugin = (plugin) => {
+  const formattedPlugin = Object.keys(plugin).reduce(
+    (pluginToFormat, key) => {
+      // TODO: Skipping authors for now as they'd already be in Sanity and there in the plugins.json file it appears to be a github username, but in Sanity it's a person's name.
+      switch (key) {
+        case 'author':
+          break
+
+        case 'compatibility':
+          // In Sanity, the compatibility field is an array of strings, but in plugins.json it's an array of objects.
+          // eslint-disable-next-line no-param-reassign
+          pluginToFormat[pluginKeyToSanityFieldNameLookup[key]] = plugin[key] ? plugin[key].map(JSON.stringify) : null
```

Comment:
@estephinson @hexagoncircle, is a JSON format stored in the CMS fine for use in the Integrations Hub, or should this be an object shaped like 

https://github.com/netlify/plugins/blob/905be79919637898b6ce3887ea41a6c4e9346378/types/plugins.d.ts#L18-L26

Reply:
The [current format for `compatibility`](https://github.com/netlify/marketing-cms/blob/main/schemas/integrations/build-plugin.js#L144-L152) will need to be updated to match this type. Right now it's only an array of strings. Should this be changed to an array of objects that contain `version`, `migrationGuide`, and `nodeVersion` fields?

Reply:
@hexagoncircle, yeah. There is always a `version` field, but for the others, it's either `migrationGuide` or `nodeVersion`. From what I've seen in plugins.json, it's never both. Maybe someone more familiar with that repo could confirm.

Reply:
Waiting for https://github.com/netlify/marketing-cms/pull/196 to be merged and then I will change this to

```suggestion
          pluginToFormat[pluginKeyToSanityFieldNameLookup[key]] = plugin[key] ?? null
```